### PR TITLE
Modification

### PR DESCRIPTION
**Pull request by Thomas905**

### CODE REVIEW

**Review by mcbeloy — DISMISSED** [review]

**Review by ghost — DISMISSED** [review]

File: src/Model/BooksManager.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        return $statement->execute();
+    }
+
+    public function testInput($data)
+    {
+        $data = trim($data);
+        $data = stripslashes($data);
+        $data = htmlspecialchars($data);
+        return $data;
+    }
```
Comment (RomainClair) [inline src/Model/BooksManager.php]: This clearly does not belong in a manager class

**Review by RomainClair — CHANGES_REQUESTED** [review]
You should not change the AbstractManager (unless you find a bug in it) because it's your framework.
Furthermore, because you removed selectAll and selectOneById, you had to recreate them all many times in every *Manager class and that probably is uncessary work.
The best practice is to use singular for SQL tables (author, editor, ...) and to have the same for classes (Author, Editor, therefore AuthorManager, EditorManager...)

File: src/Model/AbstractManager.php [hunk 2]
```diff
@@ -65,4 +65,25 @@ public function delete(int $id): void
...
+        AS status_name
+        FROM book
+        JOIN editor ON book.editor_id = editor.id
+        JOIN category ON book.category_id = category.id
+        JOIN format ON Book.format_id = format.id
+        JOIN location ON book.location_id = location.id
+        JOIN author ON book.author_id = author.id
+        JOIN status ON book.status_id = status.id';
+        return $this->pdo->query($statement)->fetchAll(PDO::FETCH_ASSOC);
+    }
```
Comment (RomainClair) [inline src/Model/AbstractManager.php]: This method should probably be in the bookManager because it return a book

**Review by RomainClair — CHANGES_REQUESTED** [review]
We probably shouldn't change the abstractManager because it's our framework.

**Review by ghost — DISMISSED** [review]
ok for me

File: src/routes.php [hunk 3]
```diff
@@ -8,5 +8,6 @@
 // e.g route '/item/edit?id=1' will execute $itemController->edit(1)
 
 return [
-    '' => ['HomeController', 'index',]
+    'book/edit' => ['BookController', 'edit', ['id']],
+    '' => ['BookController', 'index',]
```
Comment (alexandregrue) [inline src/routes.php]: Invert line 11 and 12

File: src/Model/BookManager.php [hunk 4]
```diff
@@ -2,9 +2,76 @@
 
 namespace App\Model;
 
-use App\Model\AbstractManager;
+use PDO;
```
Comment (alexandregrue) [inline src/Model/BookManager.php]: Book extends of Abstract manager so don't need use PDO

**Review by alexandregrue — CHANGES_REQUESTED** [review]

**Review by alexandregrue — DISMISSED** [review]

File: src/Controller/BookController.php [hunk 5]
```diff
@@ -3,8 +3,58 @@
...
+        $locations = $locationManager->selectAll();
+
+        $statusManager = new StatusManager();
+        $status = $statusManager->selectAll();
+
+        if ($_SERVER['REQUEST_METHOD'] === 'POST') {
+            $book = array_map('trim', $_POST);
+            $bookManager->update($book);
+            header('Location: /');
+        }
```
Comment (RomainClair) [inline src/Controller/BookController.php]: There's nearly no user input verification...

**Review by RomainClair — CHANGES_REQUESTED** [review]

**Comment by go37** [comment]
In BookManager.php, change "Book" by "book" in line 24 and 72.

File: src/Model/BookManager.php [hunk 6]
```diff
@@ -3,8 +3,76 @@
...
+        editor.id AS editor_id, editor.name AS editor_name, 
+        category.id AS category_id, category.name AS category_name, 
+        format.id AS format_id, format.name AS format_name, 
+        location.id AS location_id, location.name AS location_name, 
+        author.id AS author_id, author.name AS author_name, 
+        status.id AS status_id, status.name AS status_name
+        FROM book 
+        JOIN editor ON book.editor_id = editor.id 
+        JOIN category ON book.category_id = category.id 
+        JOIN format ON Book.format_id = format.id
```
Comment (ghost) [inline src/Model/BookManager.php]: book.format_id

File: src/Model/BookManager.php [hunk 7]
```diff
@@ -3,8 +3,76 @@
...
+        AS editor_name, category.name
+        AS category_name, format.name
+        AS format_name, location.name
+        AS location_name, author.name
+        AS author_name, status.name
+        AS status_name
+        FROM book
+        JOIN editor ON book.editor_id = editor.id
+        JOIN category ON book.category_id = category.id
+        JOIN format ON Book.format_id = format.id
```
Comment (ghost) [inline src/Model/BookManager.php]: book.format_id

**Review by ghost — DISMISSED** [review]
good

**Review by ghost — DISMISSED** [review]
ok for me

File: src/Model/BookManager.php [hunk 8]
```diff
@@ -8,6 +8,7 @@
 class BookManager extends AbstractManager
 {
     public const TABLE = "book";
+
     /* Insert element aboot the book into bdd */
```
Comment (mcbeloy) [inline src/Model/BookManager.php]: "aboot" is "about"...

**Review by mcbeloy — APPROVED** [review]

**Review by mcbeloy — APPROVED** [review]

**Review by mcbeloy — DISMISSED** [review]

**Review by mcbeloy — APPROVED** [review]

**Review by mcbeloy — APPROVED** [review]

**Review by mcbeloy — APPROVED** [review]

**Review by RomainClair — APPROVED** [review]
The user's input checks (form) are still not sufficient enough

**Pull request merged**